lib/rust/transformer/src/main.rs

```diff
@@ -402,12 +402,39 @@ async fn read_events_s3<'a>(
                 }
             })
         }
-        None => FramedRead::new(reader, LinesCodec::new())
-            .map(|v| match v {
-                Ok(v) => Ok(Value::from(v)),
-                Err(e) => Err(anyhow!(e)),
-            })
-            .boxed(),
+        None => {
+            if r.key.clone().ends_with("csv") {
+                let mut csv_reader = AsyncReader::from_reader(reader);
+                let mut headers: StringRecord = StringRecord::new();
+                { headers = csv_reader.headers().await?.to_owned(); } // do better here
+                let records = csv_reader.into_records();
```

Comment:
- I think we can start with only supporting CSV files with a header row as the first row. You can use `csv_async::AsyncReaderBuilder::new().has_headers(true)` to tell it read the first row as headers.
- You can use an `AsyncDeserializer` to automatically deserialize the values into a serde compatible type so you don't have to manually construct the HashMap from a `StringRecord`. I believe its `create_deserializer()` on the `AsyncReaderBuilder` and you can specify the type to deserialize to like `.deserialize::<HashMap<String, serde_json::Value>>()`.
- I think you have to deserialize to HashMap first since iirc deserializing to serde_json::Value  doesn't work with csv directly (iirc lmk if otherwise).

Reply:
That is correct for bullet #3, it only returns the first value in the csv file when deserializing to serde_json::Value